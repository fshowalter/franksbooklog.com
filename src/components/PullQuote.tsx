import type { JSX } from "react";

import { ccn } from "~/utils/concatClassNames";

export function PullQuote({
  attribution,
  children,
  className,
}: {
  attribution?: string;
  children: React.ReactNode;
  className?: string;
}): JSX.Element {
  return (
    // eslint-disable-next-line better-tailwindcss/no-unregistered-classes
    <blockquote className={ccn("pull-quote", className)}>
      <span className="relative z-10 italic">{children}</span>
      {attribution && (
        <cite className="mt-4 block text-base font-normal text-muted not-italic">
          â€” {attribution}
        </cite>
      )}
    </blockquote>
  );
}
