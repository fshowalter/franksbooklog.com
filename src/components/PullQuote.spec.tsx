import { render } from "@testing-library/react";
import { describe, it } from "vitest";

import { PullQuote } from "./PullQuote";

describe("PullQuote", () => {
  it("renders pull quote with text", ({ expect }) => {
    const { container } = render(
      <PullQuote>This is a memorable quote from the book.</PullQuote>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders pull quote with attribution", ({ expect }) => {
    const { container } = render(
      <PullQuote attribution="David Hackworth">
        There are only two things in life that interest me. One is sex and the
        other is adventure.
      </PullQuote>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders pull quote with custom className", ({ expect }) => {
    const { container } = render(
      <PullQuote className="text-center">
        An organization does well only those things the Boss checks.
      </PullQuote>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
