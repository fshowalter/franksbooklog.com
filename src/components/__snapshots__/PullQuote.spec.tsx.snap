// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PullQuote > renders pull quote with attribution 1`] = `
<blockquote
  class="pull-quote"
>
  <span
    class="relative z-10 italic"
  >
    There are only two things in life that interest me. One is sex and the other is adventure.
  </span>
  <cite
    class="mt-4 block text-base font-normal text-muted not-italic"
  >
    â€” 
    David Hackworth
  </cite>
</blockquote>
`;

exports[`PullQuote > renders pull quote with custom className 1`] = `
<blockquote
  class="pull-quote text-center"
>
  <span
    class="relative z-10 italic"
  >
    An organization does well only those things the Boss checks.
  </span>
</blockquote>
`;

exports[`PullQuote > renders pull quote with text 1`] = `
<blockquote
  class="pull-quote"
>
  <span
    class="relative z-10 italic"
  >
    This is a memorable quote from the book.
  </span>
</blockquote>
`;
